mlflow/models/evaluation/default_evaluator.py

```diff
@@ -171,12 +174,26 @@ def _extract_predict_fn(
         try:
             import xgboost
 
-            if isinstance(raw_model, xgboost.XGBModel):
+            def xgboost_predict_fn(raw_model):
                 # Because shap evaluation will pass evaluation data in ndarray format
                 # (without feature names), if set validate_features=True it will raise error.
-                predict_fn = partial(predict_fn, validate_features=False)
-                if predict_proba_fn is not None:
-                    predict_proba_fn = partial(predict_proba_fn, validate_features=False)
+                if isinstance(raw_model, xgboost.Booster):
+                    # we need to wrap the predict function to accept data in pandas format
+                    def wrapped_predict_fn(data, *args, **kwargs):
+                        return raw_model.predict(
+                            xgboost.DMatrix(data), *args, validate_features=False, **kwargs
+                        )
+
+                    return wrapped_predict_fn
+                elif isinstance(raw_model, xgboost.XGBModel):
+                    return partial(raw_model.predict, validate_features=False)
+                else:
+                    return raw_model.predict
```

Comment:
With this https://github.com/mlflow/mlflow/pull/12599/files#r1668422877, we can move this part code out of mlflow/models/evaluation/default_evaluator.py , to make the code easier to maintain.

Reply:
Updated according to above comment